botnim/vector_store/search_modes.py

```diff
@@ -0,0 +1,183 @@
+from typing import Dict
+from types import MappingProxyType
+from .search_config import SearchModeConfig, SearchFieldConfig, FieldWeight
+
+# Define the TAKANON_SECTION_NUMBER mode config
+TAKANON_SECTION_NUMBER_CONFIG = SearchModeConfig(
+    name="TAKANON_SECTION_NUMBER",
+    description="Specialized search mode for finding Takanon sections by their number (e.g. 'סעיף 12'). Requires both section number and resource name. The resource name can be provided in a flexible format (e.g. 'חוק הכנסת' or 'חוק-הכנסת').",
+    min_score=0.5,
+    num_results=3,  # Default for section/resource search
+    use_vector_search=True,
```

Comment:
are we sure we want to enable vector store here?
seems to me that when searching 'סעיף 777' we want to return "no results" if no text match was made